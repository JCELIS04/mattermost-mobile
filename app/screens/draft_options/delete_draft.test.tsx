// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {Screens} from '@constants';
import {dismissBottomSheet} from '@screens/navigation';
import {fireEvent, renderWithIntlAndTheme} from '@test/intl-test-helper';

import DeleteDraft from './delete_draft';

jest.mock('@screens/navigation', () => ({
    dismissBottomSheet: jest.fn(),
}));

describe('DeleteDraft', () => {
    it('should render the component', () => {
        const wrapper = renderWithIntlAndTheme(
            <DeleteDraft
                bottomSheetId={Screens.DRAFT_OPTIONS}
                channelId='channel_id'
                rootId='root_id'
            />,
        );
        const {getByText, getByTestId} = wrapper;
        expect(getByText('Delete draft')).toBeTruthy();
        expect(getByTestId('trash-can-outline-icon')).toBeTruthy();
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('calls draftDeleteHandler when pressed', () => {
        const wrapper = renderWithIntlAndTheme(
            <DeleteDraft
                bottomSheetId={Screens.DRAFT_OPTIONS}
                channelId='channel_id'
                rootId='root_id'
            />,
        );
        const {getByTestId} = wrapper;
        fireEvent.press(getByTestId('trash-can-outline-icon'));
        expect(dismissBottomSheet).toHaveBeenCalledTimes(1);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
