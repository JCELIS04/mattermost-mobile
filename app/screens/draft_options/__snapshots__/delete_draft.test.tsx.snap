// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDraft calls draftDeleteHandler when pressed 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 16,
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
  testID="delete_draft"
>
  <Icon
    color="rgba(63,67,80,0.56)"
    name="trash-can-outline"
    size={24}
    testID="trash-can-outline-icon"
  />
  <Text
    style={
      {
        "color": "#3f4350",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      }
    }
  >
    Delete draft
  </Text>
</View>
`;

exports[`DeleteDraft should render the component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 16,
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
  testID="delete_draft"
>
  <Icon
    color="rgba(63,67,80,0.56)"
    name="trash-can-outline"
    size={24}
    testID="trash-can-outline-icon"
  />
  <Text
    style={
      {
        "color": "#3f4350",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      }
    }
  >
    Delete draft
  </Text>
</View>
`;
